pkgs/coverage/test/test_with_coverage_test.dart

```diff
@@ -122,6 +124,60 @@ dependency_overrides:
       ['pub', 'global', 'run', 'coverage:test_with_coverage'],
     );
   });
+
+  test(
+      'dart run bin/test_with_coverage.dart --fail-under succeeds when coverage meets threshold',
+      () async {
+    final process = await _run([
+      'run',
+      _testWithCoveragePath,
+      '--fail-under=100', // This should pass as coverage=100% when all tests run
```

Comment:
You've got a few of these comments throughout this test, some of which are multi-line. Can you move them to the start of each test. Eg, this comment should be inserted between line 130 and 131. That'll make the multi-line comments easier to read.